src/ControllerStatusIO.c

```diff
@@ -114,23 +120,28 @@ BOOL Ros_Controller_Initialize()
     }
 
     //get the robot calibration data for multi-robot systems
-    for (i = 0; i < MAX_ROBOT_CALIBRATION_FILES; i += 1)
+    const BOOL bCalibLoadedOk = Ros_Controller_LoadGroupCalibrationData(&g_Ros_Controller);
+    //see whether the user should be notified about failures (it's OK to not
+    //have/load any calibration in some cases)
+    const BOOL bNeedToWarnAboutCalib = Ros_Controller_ShouldWarnNoCalibDataLoaded(
+        &g_Ros_Controller, g_nodeConfigSettings.publish_tf, bCalibLoadedOk);
+    Ros_Debug_BroadcastMsg(
+        "%s: calib loaded ok: %s, should warn: %s", __func__,
+            bCalibLoadedOk ? "yes" : "no", bNeedToWarnAboutCalib ? "yes" : "no");
+    if (bNeedToWarnAboutCalib)
     {
-        MP_RB_CALIB_DATA calibData;
-        if (Ros_mpGetRobotCalibrationData(i, &calibData) == OK)
+        if (g_nodeConfigSettings.ignore_missing_calib_data)
         {
-            if (calibData.s_rb.grp_no <= MP_R8_GID && //the slave is a robot
-                calibData.m_rb.grp_no <= MP_R8_GID) //the master is another robot's RF
-            {
-                groupIndex = mpCtrlGrpId2GrpNo((MP_GRP_ID_TYPE)calibData.s_rb.grp_no);
-                MP_COORD* coord = &g_Ros_Controller.ctrlGroups[groupIndex]->robotCalibrationToBaseFrame;
-                coord->x = calibData.pos_uow[0];
-                coord->y = calibData.pos_uow[1];
-                coord->z = calibData.pos_uow[2];
-                coord->rx = calibData.ang_uow[0];
-                coord->ry = calibData.ang_uow[1];
-                coord->rz = calibData.ang_uow[2];
-            }
+            Ros_Debug_BroadcastMsg(
+                "%s: ignoring absence of calibration data (or load failure) as configured");
+        }
+        else
+        {
+            mpSetAlarm(ALARM_CONFIGURATION_FAIL, "No calibration: check TF",
+                SUBCODE_CONFIGURATION_NO_CALIB_FILES_LOADED);
+            Ros_Debug_BroadcastMsg(
+                "%s: no calibration files loaded: TF potentially incorrect -> warn "
```

Comment:
To me, the `-> warn` is confusing. I'd suggest just doing something like `warning: no calibration...`

Reply:
I'll rephrase it, but this is logging whether a warning should be displayed. The `->` is a logic implication.